<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>523</width>
    <height>671</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>PID channel settings tab</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Sweep</string>
     </property>
     <layout class="QFormLayout" name="formLayout_2">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Center</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QSlider" name="centerSlider">
          <property name="minimum">
           <number>-32768</number>
          </property>
          <property name="maximum">
           <number>32767</number>
          </property>
          <property name="pageStep">
           <number>1000</number>
          </property>
          <property name="value">
           <number>0</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="centerSpinBox">
          <property name="minimum">
           <number>-32768</number>
          </property>
          <property name="maximum">
           <number>32767</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Range</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QSlider" name="rangeSlider">
          <property name="maximum">
           <number>65535</number>
          </property>
          <property name="pageStep">
           <number>1000</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="rangeSpinBox">
          <property name="maximum">
           <number>65535</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Frequency</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QSlider" name="frequencySlider">
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="pageStep">
           <number>25</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="frequencySpinBox">
          <property name="maximum">
           <number>1000</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="3" column="1">
       <widget class="QPushButton" name="sweepButton">
        <property name="text">
         <string>Run Sweep</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>PID Controller</string>
     </property>
     <layout class="QFormLayout" name="formLayout_3">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>P Gain</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QSlider" name="pGainSlider">
          <property name="maximum">
           <number>65535</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="pGainSpinBox">
          <property name="maximum">
           <number>65535</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>I Gain</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <item>
         <widget class="QSlider" name="iGainSlider">
          <property name="maximum">
           <number>65535</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="iGainSpinBox">
          <property name="maximum">
           <number>65535</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>Input Offset</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_8">
        <item>
         <widget class="QSlider" name="inputOffsetSlider">
          <property name="minimum">
           <number>-512</number>
          </property>
          <property name="maximum">
           <number>511</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="inputOffsetSpinBox">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimum">
           <number>-512</number>
          </property>
          <property name="maximum">
           <number>511</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_11">
        <property name="text">
         <string>Output Offset</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_9">
        <item>
         <widget class="QSlider" name="outputOffsetSlider">
          <property name="minimum">
           <number>-32768</number>
          </property>
          <property name="maximum">
           <number>32767</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="outputOffsetSpinBox">
          <property name="minimum">
           <number>-32768</number>
          </property>
          <property name="maximum">
           <number>32767</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="label_12">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_10">
        <item>
         <widget class="QPushButton" name="resetPidButton">
          <property name="text">
           <string>Reset PID Controller</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="flipPolarityButton">
          <property name="text">
           <string>Flip Polarity</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_13">
        <property name="text">
         <string>D Gain</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_12">
        <item>
         <widget class="QSlider" name="dGainSlider">
          <property name="maximum">
           <number>65535</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="dGainSpinBox">
          <property name="maximum">
           <number>65535</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_4">
     <property name="title">
      <string>Automatic Relocking</string>
     </property>
     <layout class="QFormLayout" name="formLayout_4">
      <item row="0" column="0">
       <widget class="QLabel" name="label_16">
        <property name="text">
         <string>Filter Response</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_17">
        <property name="text">
         <string>Threshold</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_14">
        <item>
         <widget class="QSlider" name="filterResponseSlider">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>32767</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="filterResponseSpinBox">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>32767</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="2" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_15">
        <item>
         <widget class="QSlider" name="thresholdSlider">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>511</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="thresholdSpinBox">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>511</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="4" column="1">
       <widget class="QCheckBox" name="relockingEnabledCheckBox">
        <property name="text">
         <string>Enable</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QSlider" name="ttlExpSlider">
          <property name="maximum">
           <number>65535</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="TTLExpansionSpinBox" name="ttlExpSpinBox">
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
          <property name="suffix">
           <string> ms</string>
          </property>
          <property name="maximum">
           <number>65535</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_18">
        <property name="text">
         <string>TTL Expansion</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>TTLExpansionSpinBox</class>
   <extends>QSpinBox</extends>
   <header>evil.customwidgets</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>centerSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>centerSpinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>304</x>
     <y>44</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>49</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rangeSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>rangeSpinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>415</x>
     <y>73</y>
    </hint>
    <hint type="destinationlabel">
     <x>486</x>
     <y>73</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>frequencySlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>frequencySpinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>430</x>
     <y>103</y>
    </hint>
    <hint type="destinationlabel">
     <x>488</x>
     <y>100</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pGainSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>pGainSpinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>401</x>
     <y>204</y>
    </hint>
    <hint type="destinationlabel">
     <x>468</x>
     <y>193</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>iGainSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>iGainSpinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>420</x>
     <y>227</y>
    </hint>
    <hint type="destinationlabel">
     <x>475</x>
     <y>225</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dGainSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>dGainSpinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>406</x>
     <y>252</y>
    </hint>
    <hint type="destinationlabel">
     <x>472</x>
     <y>252</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>inputOffsetSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>inputOffsetSpinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>419</x>
     <y>276</y>
    </hint>
    <hint type="destinationlabel">
     <x>463</x>
     <y>283</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>outputOffsetSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>outputOffsetSpinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>425</x>
     <y>306</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>312</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>filterResponseSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>filterResponseSpinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>420</x>
     <y>414</y>
    </hint>
    <hint type="destinationlabel">
     <x>460</x>
     <y>410</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>thresholdSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>thresholdSpinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>392</x>
     <y>433</y>
    </hint>
    <hint type="destinationlabel">
     <x>465</x>
     <y>442</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ttlExpSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>ttlExpSpinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>365</x>
     <y>464</y>
    </hint>
    <hint type="destinationlabel">
     <x>447</x>
     <y>465</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>centerSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>centerSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>457</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>370</x>
     <y>50</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rangeSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>rangeSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>460</x>
     <y>70</y>
    </hint>
    <hint type="destinationlabel">
     <x>397</x>
     <y>78</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>frequencySpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>frequencySlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>461</x>
     <y>100</y>
    </hint>
    <hint type="destinationlabel">
     <x>364</x>
     <y>110</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pGainSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>pGainSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>480</x>
     <y>201</y>
    </hint>
    <hint type="destinationlabel">
     <x>401</x>
     <y>201</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>iGainSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>iGainSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>462</x>
     <y>219</y>
    </hint>
    <hint type="destinationlabel">
     <x>423</x>
     <y>227</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dGainSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>dGainSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>458</x>
     <y>259</y>
    </hint>
    <hint type="destinationlabel">
     <x>373</x>
     <y>259</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>inputOffsetSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>inputOffsetSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>477</x>
     <y>286</y>
    </hint>
    <hint type="destinationlabel">
     <x>411</x>
     <y>284</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>outputOffsetSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>outputOffsetSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>459</x>
     <y>308</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>318</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>filterResponseSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>filterResponseSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>464</x>
     <y>398</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>408</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>thresholdSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>thresholdSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>475</x>
     <y>432</y>
    </hint>
    <hint type="destinationlabel">
     <x>390</x>
     <y>433</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ttlExpSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>ttlExpSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>432</x>
     <y>474</y>
    </hint>
    <hint type="destinationlabel">
     <x>371</x>
     <y>472</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
